core/js/src/main/scala/org/log4s/log4sjs/FunctionalType.scala

```diff
@@ -9,8 +9,12 @@ abstract class FunctionalType[ObjectType <: js.Any, Argument, Result]
   type FunctionType = js.Function1[Argument, Result]
   type DynamicType = js.|[FunctionType, Type]
 
+  // Workaround for formatter.merge.asInstanceOf[js.Dynamic] since formatter.merge doesn't work on Scala 3.0.0
+  private def unionMerge[A <: js.|[_, _], B](a: A)(implicit ev: js.|.Evidence[A, B]): B =
+    a.asInstanceOf[B]
```

Comment:
Since Scala.js linking process doesn't find `formatter.merge` which is an extension method from `UnionOps`, I wrote this `unionMerge` method here, and it does what `UnionOps.merge` does. If I'm wrong, please let me know. I'm not a Scala.js user so I'm so sure about this.

For your convenience, this is `UnionOps` from Scala.js.
```scala
  implicit class UnionOps[A <: _ | _] private[|] (private val self: A)
      extends AnyVal {

    /** Explicitly merge a union type to a supertype (which might not be a
     *  union type itself).
     *
     *  This needs evidence that `A <: B`.
     */
    def merge[B](implicit ev: |.Evidence[A, B]): B =
      self.asInstanceOf[B]
  }
```